babel/dates.py

```diff
@@ -1505,8 +1503,15 @@ def get_week_number(self, day_of_period, day_of_week=None):
         if first_day < 0:
             first_day += 7
         week_number = (day_of_period + first_day - 1) // 7
+
         if 7 - first_day >= self.locale.min_week_days:
             week_number += 1
+
+        if self.locale.first_week_day == 0:
```

Comment:
I'd love to see a comment explaining the rationale here, since the code (like all date code) is a little obtuse anyway :)